tests/gen/schema_parser.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package gen
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/request/graphql"
+)
+
+func parseSchema(schema string) (map[string]client.CollectionDefinition, error) {
+	parser, err := graphql.NewParser()
+	if err != nil {
+		return nil, err
+	}
+	cols, err := parser.ParseSDL(context.Background(), schema)
+	if err != nil {
+		return nil, err
+	}
+	result := make(map[string]client.CollectionDefinition)
+	for _, col := range cols {
+		result[col.Description.Name] = col
+	}
+	return result, nil
+}
+
+func parseConfig(schema string) (configsMap, error) {
+	genConfigs := make(map[string]map[string]genConfig)
+
+	var currentConfig map[string]genConfig
+	var currentType string
+
+	schemaLines := strings.Split(schema, "\n")
+	for _, line := range schemaLines {
+		line = strings.TrimSpace(line)
+		if strings.HasPrefix(line, "type ") {
+			typeNameEndPos := strings.Index(line[5:], " ")
```

Comment:
todo: It looks like `5` is used because that is how many character are in the string `type `, however readers are left to guess this.  Can you please extract `5` to a variable, maybe:
```
typePrefixLen := len("type ")
```